examples/chat_app.md

```diff
@@ -0,0 +1,251 @@
+# Building a chat app with Deno
+
+## Introduction
+
+In this tutorial we'll create a simple chat app using Deno. Our chat app will
+allow multiple chat clients connected to the same backend to send group messages
+through web sockets. After a client chooses a username, they can then start
+sending group messages to other online clients. Each client also displays the
+list of currently active users.
+
+![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/4002626d-edc6-41dd-a518-c6197df4f450/Untitled.png)
+
+## Building the View
+
+We can build the simple UI shown above with the following as our `index.html`.
+Note that:
+
+1. We're using Bootstrap for styling
+2. The `app.js` script is our chat client (which will be discussed in detail
+   later)
+
+```html
+<!-- index.html -->
+
+<html>
+  <head>
+    <title>Node Chat App</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/css/bootstrap.min.css"
+    />
+    <script src="/public/app.js"></script>
+  </head>
+  <body>
+    <div class="navbar navbar-inverse navbar-fixed-top">
+      <div class="container">
+        <div class="collapse navbar-collapse">
+          <ul class="nav navbar-nav"></ul>
+        </div>
+      </div>
+    </div>
+
+    <div class="container">
+      <div id="dashboard">
+        <div class="row" />
+        <div class="col-md-4">
+          <b>Users</b>
+          <hr />
+          <div id="users"></div>
+          <hr class="visible-xs visible-sm" />
+        </div>
+        <div class="col-md-8">
+          <input
+            class="col-md-12 col-sm-12"
+            id="data"
+            placeholder="send message"
+          />
+          <hr />
+          <div id="conversation"></div>
+        </div>
+      </div>
+    </div>
+  </body>
+</html>
+```
+
+## **WebSocket** Primer
+
+We will rely on Deno's native support for web sockets when building our client
+and server. A [web socket](https://en.wikipedia.org/wiki/WebSocket) is a
```

Comment:
MDN is a much better source for stuff like this: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket